classes/local/skel/lib_php_file.php

```diff
@@ -66,4 +66,26 @@ public function add_supported_feature($feature) {
 
         $this->data['self']['supports'][] = $feature;
     }
+
+    /**
+     * Returns the variables needed for rendering the template.
+     *
+     * @param string $plugintype.
+     * @return string[].
+     */
+    static public function get_template_variables($plugintype = null) {
+
+        $templatevars = array();
+
+        if ($plugintype == 'enrol') {
+            $templatevars = array(
+                array('name' => 'allow_enrol', 'hint' => 'boolean', 'values' => array('True', 'False')),
```

Comment:
You're right, I've already redesigned that [here](https://github.com/mudrd8mz/moodle-tool_pluginskel/pull/59/files#diff-031b6f560b174176685803de7a4de000R116). It's now just a hint for the UI, like you're said.

If I want to have some sort of strict checking for the variable values, I can pass that as an option to the manager. but this is not a priority right now.

An example of usage is in the [classes/step1_form.php](https://github.com/mudrd8mz/moodle-tool_pluginskel/pull/59/files#diff-ea1e452978d6ba0456fcdb26071c22f7R140) file. 

I'll update this pull request.